fastlane/fastlane
# Extend update_plist documentation from 1 to 6 code-examples

### Checklist

### Motivation and Context
It took me a considerable amount of time and trial & error to initially figure out how to actually use the `update_plist` action to modify the values inside a plist-file, whose functionality was (imho) only sparsely documented.

### Description
I added five new examples for the basic functionality of this action in the documentation, in addition to the one (more complicated and highly specific) example already present, to improve understandability and readability in the future.

I also slightly adapted `self.details` in order to differentiate this action from the (similar but different in usage) `update_info_plist` action.

### Testing Steps
In addition to running `bundle exec rspec` and `bundle exec rubocop -a` succesfully I tested all of the example-codes and checked the plist-file in XCode afterwards to make sure the values were correctly set. See encompassing testing-code and result-image below:

```
lane :test_update_plist do |options|
  options[:environment] = "development"

  # Updates values in Firebase-plist
  update_plist(
    plist_path: "GoogleService-Info.plist",
    block: proc do |plist|
      if options[:environment] == "production"
        plist[:boolean_entry] = true
        plist[:number_entry] = 13
        plist[:array_entry] = ["entry_01", true, 1243]
      else
        plist[:boolean_entry] = false
        plist[:number_entry] = 27
        plist[:array_entry] = ["entry_02", false, 4265]
      end
    end
  )
end
```
![update_plist_test_results](https://user-images.githubusercontent.com/2097526/71891100-1d768600-3146-11ea-9a45-3cb0b74e1603.png)
